query_optimizer/ExecutionGenerator.cpp

```diff
@@ -670,10 +670,18 @@ void ExecutionGenerator::convertHashJoin(const P::HashJoinPtr &physical_plan) {
   }
 
   // Create join hash table proto.
-  const QueryContext::join_hash_table_id join_hash_table_index =
-      query_context_proto_->join_hash_tables_size();
+  std::vector<QueryContext::join_hash_table_id> join_hash_table_index;
+
   S::HashTable *hash_table_proto = query_context_proto_->add_join_hash_tables();
 
+  const PartitionScheme &part_scheme = build_relation_info->relation->getPartitionScheme();
+  const std::size_t num_partitions = part_scheme.getPartitionSchemeHeader().getNumPartitions();
```

Comment:
We need to check whether `partition_scheme` exists using `hasPartitionScheme` before using it. Otherwise, we will encounter segfault.